mango-core/src/main/scala/org/bdgenomics/mango/models/GenotypeMaterialization.scala

```diff
@@ -79,6 +80,47 @@ class GenotypeMaterialization(s: SparkContext,
     // write variants and genotypes to json
     genotypes.mapValues(v => write(v))
   }
+
+  /**
+   * Formats raw data from RDD to JSON.
+   *
+   * @param region Region to obtain coverage for
+   * @param binning Tells what granularity of coverage to return. Used for large regions
+   * @return JSONified data map
+   */
+  def getGenotype(region: ReferenceRegion, binning: Int = 1): Map[String, String] = {
+    val data: RDD[(String, Genotype)] = get(region)
+    val flattened: Map[String, Array[(String, VariantJson)]] = data
+      .map(r => {
+        val geno = r._2
+        val bin: Long = geno.getStart - (geno.getStart % binning)
```

Comment:
Bin should still be the same datatype as the return type of Genotype.getStart, but I doubt it would overflow int. I will switch it to int for now and revert back if necessary.